### DISCUSSION TITLE
Welcome to FSM Discussions!

### DISCUSSION THREAD

**Discussion by maxekman** [looplab/fsm#80] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by majst01** [comment]
Since go-1.18 we have generics, it would be worth at least to have a open discussion if it would be a valuable addition to fsm to change from "string" events and states to generic `events E` and `states S`. I could also try to create a POC PR therefore.

**Reply by maxekman** [reply]
Sounds like a good idea! I don’t currently use the library myself in an active project, but would love to see a POC.

**Reply by majst01** [reply]
https://github.com/looplab/fsm/pull/86
